### PR TITLE
Fix recipe check

### PR DESCRIPTION
**Pull request by HoleFish**
it's an attempt to fix broken recipes that need "specific" circuits. need more tests. closes [#14935](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/14935).

### CODE REVIEW

File: src/main/java/gregtech/api/util/GT_Recipe.java [hunk 1]
```diff
@@ -802,27 +805,43 @@ public double maxParallelCalculatedByInputs(int maxParallel, FluidStack[] aFluid
...
                 if (itemStack == null) continue;
+                if (isNBTSensitive) {
+                    itemMap.merge(GT_Utility.ItemId.createNoCopy(itemStack), itemStack.stackSize, Integer::sum);
+                } else {
+                    itemMap.merge(GT_Utility.ItemId.createWithoutNBT(itemStack), itemStack.stackSize, Integer::sum);
+                }
+                if (foundWildcard) {
+                    itemMapWildcard
+                        .merge(GT_Utility.ItemId.createAsWildcard(itemStack), itemStack.stackSize, Integer::sum);
+                }
```
Comment (miozune) [inline src/main/java/gregtech/api/util/GT_Recipe.java:816]: nit: Merging if conditions might be better, like
```java
if (isNBTSensitive) {
    itemMap.merge(GT_Utility.ItemId.createNoCopy(itemStack), itemStack.stackSize, Integer::sum);
    itemMapUnified
        .merge(GT_Utility.ItemId.createNoCopy(unifiedStack), unifiedStack.stackSize, Integer::sum);
```

**Review by miozune — COMMENTED** [review]
While this fix should work for almost all the cases, there might be a few edge cases: A recipe that has both "unification target" item and its "subset oredict" item, e.g. "any HV circuit" and specific HV circuit / oak plank and darkoak plank / fish and salmon
Throwing runtime exception when such a recipe is found would be nice to prevent accidental discount of the recipes.

**Comment by Dream-Master** [comment]
Seems to work. Maybe we need to test more .

**Comment by miozune** [comment]
Now I have different idea. How about changing the logic back to the one looping over inputs instead of using hashmap?
- As I stated, in theory there would be some kind of recipes that will behave in unintended way. Even though we might not have such a recipe in the modpack, this fact prevents adding one in the future.
- If we ever decide to refactor recipe system to allow dynamic recipe matching (e.g. transform output NBT depending on input), hashmap strategy won't work.
- This causes performance regression, but I guess Θ(N+M) -> Θ(NM) isn't that much here, given the size of N and M (recipe input size N usually <= 9, machine input size M <= 16 or 32 for stocking bus.) Your improvement on `GT_ParallelHelper` not looping `isRecipeInputEqual` 128 times is already huge.

**Comment by HoleFish** [comment]
> * If we ever decide to refactor recipe system to allow dynamic recipe matching (e.g. transform output NBT depending on input), hashmap strategy won't work.

Could you explain how hashmap won't work detailedly? I think it's just calculating parallel by inputs and able to work by simply setting  `isNBTSensitive` to false.

**Comment by miozune** [comment]
Setting `isNBTSensitive` to true for one item input makes all other item inputs not being able to use oredict

**Review by miozune — APPROVED** [review]
Looks good me!

**Comment by HoleFish** [comment]
Now the second-to-last commit is useful again.

**Pull request merged**